src/topic_training/src/Begin_Here.md

```diff
@@ -0,0 +1,122 @@
+
+If you run rqt_graph on the PlotDemo(.cpp or .py) file, it should show just the name of the file (+ the ID number if you are using Python) by itself with no arrow. If you run rqt_graph when both files are running, there might be an error in the terminal, but the graph application should show both files. PlotDemo(.cpp or .py) will be by itself with no arrow, and Demo(.cpp or .py) will be by itself with or without an arrow depending on the order you executed the two files.
+
+Feel free to look through the file PlotDemo(.cpp or .py) to see how the math works and to see another example of a Publisher but with a different message type. For simplicity's sake, the comments in these files are only explaining the math since you should already understand the basic ROS functions from the previous section of this training.
+
+### Defining a Custom Topic Type
+A topic's type is determined by the type of the message being published. If you run the command *rostopic type chatter*, you should see the terminal print out "std_msgs/String" while roscore and rosrun are executing Demo(.cpp or .py). The command will print out "std_msgs/Float64" if you are executing PlotDemo(.cpp or .py).
+
+A topic can have messages of type int8, int16, int32, int64, uint*, float32, float64, String, time, duration, other msg files, and array[] of variable length or fixed length. To message a different type, just change where ever there would be a String in Demo(.cpp or .py) to your desired type and update the includes (C++) or imports (Python) at the top of the file.
+
+## Challenge
+Once you have completed the training from WRoverPlayground, you should have some working code that will be put into a simulation. The simulation will have a rover that your code is meant to control its drive through mock navigation data to reach some target/beacon. 
+
+You will take in sensor data as a 180-element array of floats that represents a 180 degree view of the robot. Each element is a degree-sized slice of the robot's view. One element will contain a float, while the other elements will be undefined. The float in this array represents the distance from the head of the robot to the head of the target. The location of the element in the array represents its position relative to the robot. If the float is in the 90th element, the target is straight ahead. If the float is before the 90th element, the target is to the robot's left, and vice versa. 
```

Comment:
I feel like this paragraph needs to be further expanded in general. There's not enough context for new members to understand what the array represents and how to interpret it. You can mention things like:
- This array is supposed to mock the input from a LIDAR which is what we use for our competition robot to add more context. Maybe even include a link to a webpage that explains what a LIDAR is. 
- The array is a series of readings of distances detected by the LIDAR. It is clockwise, with the 0th index mapping to 9 O'Clock, the 89th index mapping to 12 O'Clock, and the 189th index mapping to 3 O'Clock.
- A reading of `undefined` means that there are no obstacles detected in that direction (although from my memory the real LIDAR uses `Inf` instead of `undefined`).

Reply:
Fixed in 68e6610.